src/frontends/lean/builtin_cmds.cpp

```diff
@@ -403,13 +411,22 @@ static environment eval_cmd(parser & p) {
     auto out = p.mk_message(p.cmd_pos(), p.pos(), INFORMATION);
     out.set_caption("eval result");
     scope_traces_as_messages scope_traces(p.get_stream_name(), p.cmd_pos());
-    char * redirected; size_t redir_size;
-    FILE * fp = open_memstream(&redirected, &redir_size);
-    std::unique_ptr<char *, stream_buffer_delete> stream_buffer(&redirected);
+    object_ref pipe(lean_io_mk_pipe(box(0)));
+    if (io_result_is_error(pipe.raw())) {
```

Comment:
Have you tried to use the helper function template `get_io_result<...>`? It does the error handling for us, and convert an IO exception into a C++ exception.

Reply:
Thanks, I hadn't seen that function. I'll use it.